views/default/forms/register.php

```diff
@@ -8,6 +8,11 @@
 
 if (elgg_is_sticky_form('register')) {
 	$values = elgg_get_sticky_values('register');
+
+	// Add the sticky values to $vars so views extending
+	// register/extend also get access to them.
+	$vars = array_merge($vars, $values);
+
```

Comment:
I believe key name conflicts are very unlikely. Also we're doing similar merges also in other views.